src/components/SettingsInputText/SettingsInputText.vue

```diff
@@ -0,0 +1,201 @@
+			<p v-if="hint" class="hint">
+				{{ hint }}
+			</p>
+		</div>
+	</form>
+</template>
+
+<script>
+import GenRandomId from '../../utils/GenRandomId'
+export default {
+	name: 'SettingsInputText',
+	props: {
+		/**
+		 * label of the select group element
+		 */
+		label: {
+			type: String,
+			required: true,
+		},
+
+		/**
+		* hint of the select group input
+		*/
+		hint: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		* value of the select group input
+		 */
+		value: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * disabled state of the settings select group input
+		 */
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * id attribute of the select group element
+		 */
+		id: {
+			type: String,
+			default: () => 'settings-input-text-' + GenRandomId(),
+			validator: id => id.trim() !== '',
+		},
+	},
+	computed: {
+		/**
+		 * @returns {string}
+		 */
+		idSubmit() {
+			return this.id + '-submit'
+		},
+	},
+	methods: {
+		onInput(event) {
+			this.$emit('input', event)
+			/**
+			 * Emitted when the inputs value changes
+			 * @type {string}
+			 */
+			this.$emit('update:value', event.target.value)
+		},
+		onSubmit(event) {
+			event.preventDefault()
+			event.stopPropagation()
+			if (!this.disabled) {
+				/**
+				 * Emitted on submit of the input field
+				 * @type {Event}
+				 */
+				this.$emit('submit', event)
+			} else {
+				// ignore submit
+				return false
+			}
+		},
+		onChange(event) {
+			/**
+			 * Emitted on change of the input field
+			 * @type {Event}
+			 */
+			this.$emit('change', event)
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+
+	$input-margin: 4px;
+
+	.input-wrapper {
+		display: flex;
+		align-items: flex-start;
+		flex-wrap: wrap;
+		width: 100%;
+		max-width: 400px;
+		&__form {
+			display: flex;
+			align-items: center;
+			flex: 1 1 auto;
+
+			margin: $input-margin 0;
+			padding-right: $icon-margin;
+		}
+
+		&__submit {
+			position: absolute;
+			left: -10000px;
+			top: auto;
+			width: 1px;
+			height: 1px;
+			overflow: hidden;
+		}
+
+		&__label {
+			display: flex;
+			align-items: center;
+			justify-content: center;
+
+			width: #{$clickable-area - $input-margin * 2};
+			height: #{$clickable-area - $input-margin * 2};
+
+			cursor: pointer;
+
+			opacity: $opacity_full;
+			color: var(--color-text-lighter);
+			border: 1px solid var(--color-border-dark);
+			border-left-color: transparent;
+			border-radius: 0 var(--border-radius) var(--border-radius) 0;
+			/* Avoid background under border */
+			background-color: var(--color-main-background);
+			background-clip: padding-box;
+
+			font-size: $icon-size;
+		}
+
+		&__input[type=text] {
+			flex-grow: 1;
+			white-space: nowrap;
+			text-overflow: ellipsis;
+			overflow: hidden;
+			cursor: text;
+		}
+
+		// if disabled, change cursor
+		&:disabled {
+			cursor: default;
+		}
+
+			.hint {
+				color: var(--color-text-lighter);
+			}
+		}
```

Comment:
This is one `}` to much ;)

Reply:
thank you :), fixed. Can you please have a general look and merge it if all is good :D